src/java_bytecode/library/src/org/cprover/CProver.java

```diff
@@ -3,9 +3,124 @@
 public final class CProver
 {
   public static boolean enableAssume=true;
-  public static void assume(boolean condition) 
+  public static boolean enableNondet=true;
+
+  public static boolean nondetBoolean()
+  {
+    if (enableNondet)
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.nondetBoolean()");
+    }
+
+    return false;
+  }
+
+  public static byte nondetByte()
+  {
+    if (enableNondet)
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.nondetByte()");
+    }
+
+    return 0;
+  }
+
+  public static char nondetChar()
+  {
+    if (enableNondet)
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.nondetChar()");
+    }
+
+    return '\0';
+  }
+
+  public static short nondetShort()
+  {
+    if (enableNondet)
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.nondetShort()");
+    }
+
+    return 0;
+  }
+
+  public static int nondetInt()
+  {
+    if (enableNondet)
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.nondetInt()");
+    }
+
+    return 0;
+  }
+
+  public static long nondetLong()
+  {
+    if (enableNondet)
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.nondetLong()");
+    }
+
+    return 0;
+  }
+
+  public static float nondetFloat()
+  {
+    if (enableNondet)
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.nondetFloat()");
+    }
+
+    return 0;
+  }
+
+  public static double nondetDouble()
+  {
+    if (enableNondet)
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.nondetDouble()");
+    }
+
+    return 0;
+  }
+
+  public static <T> T nondetWithNull()
+  {
+    if (enableNondet)
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.nondetWithNull<T>(T)");
+    }
+
+    return null;
+  }
+
+  public static <T> T nondetWithoutNull()
+  {
+    if (enableNondet)
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.nondetWithoutNull<T>(T)");
+    }
+
+    return null;
+  }
+
+  public static void assume(boolean condition)
   {
     if(enableAssume)
-      throw new RuntimeException("Cannot execute program with CProver.assume()");
-  }  
+    {
+      throw new RuntimeException(
+          "Cannot execute program with CProver.assume()");
+    }
```

Comment:
Should this change be made in this PR, given that assume was added in a previous PR and has not been modified (other than formatting) in this PR?